jupyter_server/services/contents/filemanager.py

```diff
@@ -84,14 +84,33 @@ def _validate_post_save_hook(self, proposal):
             raise TraitError("post_save_hook must be callable")
         return value
 
-    def run_post_save_hook(self, model, os_path):
-        """Run the post-save hook if defined, and log errors"""
+    def register_post_save_hook(self, hook):
```

Comment:
I think this method should move to the base ContentsManager, since this is an API that all ContentsManagers should inherit, right?

Otherwise, JupyterLab, for example, will break with custom ContentsManagers inheriting from this base class.

Reply:
Yes, I don't know why the post-save hooks were in `FileContentsManager` and the pre-save hooks in `ContentsManager` though. Should everything related to save hooks be moved to `ContentsManager`?

Reply:
I think so. I'm guessing this was an oversight from the early days?